### PR TITLE
rd/aws_imagebuilder_distribution_configuration - support fastLaunchConfigurations

### LINKED ISSUE #25397: Feature Request: aws_imagebuilder_distribution_configuration support fastLaunchConfigurations
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

[CreateDistributionConfiguration](https://docs.aws.amazon.com/imagebuilder/latest/APIReference/API_CreateDistributionConfiguration.html) and [UpdateDistributionConfiguration](https://docs.aws.amazon.com/imagebuilder/latest/APIReference/API_UpdateDistributionConfiguration.html) support [`fastLaunchConfigurations`](https://docs.aws.amazon.com/imagebuilder/latest/APIReference/API_Distribution.html#imagebuilder-Type-Distribution-fastLaunchConfigurations), so the resulting Windows AMIs take advantage of [EC2 Fast Launch](https://docs.aws.amazon.com/AWSEC2/latest/WindowsGuide/win-ami-config-fast-launch.html).

### New or Affected Resource(s)

* `resource aws_imagebuilder_distribution_configuration`
* `data aws_imagebuilder_distribution_configuration`

### Potential Terraform Configuration

```hcl
resource "aws_imagebuilder_distribution_configuration" "example" {
  name = "example"
  distribution {
    ami_distribution_configuration {
      #...
    }
    launch_template_configuration {
      #...
    }
    region = "us-east-1"
    fast_launch_configuration {
      account_id = "123456789012"
      enabled    = true
      launch_template {
        #One of launch_template_id or launch_template_name required
        launch_template_id = "lt-0aaa1bcde2ff3456"
        #launch_template_name = "my_launch_template"
        launch_template_version = 2
      }
      max_parallel_launches = 5
      snapshot_configuration {
        target_resource_count = 10
      }
    }
  }
}
```

### References
* [AWS Announces New Launch Speed Optimizations for Microsoft Windows Server Instances on Amazon EC2](https://aws.amazon.com/about-aws/whats-new/2022/01/aws-speed-optimizations-windows-instances-ec2/)

### PR DESCRIPTION
**Pull request by kamilturek**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #25397.

Output from acceptance testing:

```
$ ACCTEST_PARALLELISM=2 make testacc PKG=imagebuilder TESTS="TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_"

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/imagebuilder/... -v -count 1 -parallel 2 -run='TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_'  -timeout 180m
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_enabled
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_enabled
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_launchTemplate
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_launchTemplate
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_maxParallelLaunches
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_maxParallelLaunches
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_snapshotConfiguration
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_snapshotConfiguration
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_enabled
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_maxParallelLaunches
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_maxParallelLaunches (49.51s)
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_launchTemplate
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_enabled (50.51s)
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_snapshotConfiguration
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_launchTemplate (50.20s)
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_snapshotConfiguration (56.36s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/imagebuilder       108.513s
```
```
$ make testacc PKG=imagebuilder TESTS="TestAccImageBuilderDistributionConfigurationDataSource_arn"

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/imagebuilder/... -v -count 1 -parallel 20 -run='TestAccImageBuilderDistributionConfigurationDataSource_arn'  -timeout 180m
=== RUN   TestAccImageBuilderDistributionConfigurationDataSource_arn
=== PAUSE TestAccImageBuilderDistributionConfigurationDataSource_arn
=== CONT  TestAccImageBuilderDistributionConfigurationDataSource_arn
--- PASS: TestAccImageBuilderDistributionConfigurationDataSource_arn (41.60s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/imagebuilder       43.253s
```

### CODE REVIEW

File: internal/service/imagebuilder/distribution_configuration.go [hunk 1]
```diff
@@ -168,6 +168,68 @@ func ResourceDistributionConfiguration() *schema.Resource {
...
+												"launch_template_id": {
+													Type:         schema.TypeString,
+													Optional:     true,
+													ValidateFunc: verify.ValidLaunchTemplateID,
+												},
+												"launch_template_name": {
+													Type:         schema.TypeString,
+													Optional:     true,
+													ValidateFunc: verify.ValidLaunchTemplateName,
+												},
```
Comment (kamilturek) [inline internal/service/imagebuilder/distribution_configuration.go:201]: Is there a way to use `ExactlyOneOf` in a nested set element? It cannot be done like below as the set can have more than 1 item.
```
ExactlyOneOf: []string{"distribution.0.fast_launch_configuration.0.launch_template.0.launch_template_id", ...}
```

**Comment by kamilturek** [comment]
`providerlint` failed for some unknown reason, it was running for an hour. Not sure if eveything is alright there.

File: internal/service/imagebuilder/distribution_configuration.go (hunk 1 above)
Comment (ewbankkit) [reply internal/service/imagebuilder/distribution_configuration.go:201]: Unfortunately not right now (needs the great work being done over here: https://github.com/hashicorp/terraform-plugin-framework-validators) 😞.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccImageBuilderDistributionConfiguration' PKG=imagebuilder ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/imagebuilder/... -v -count 1 -parallel 2  -run=TestAccImageBuilderDistributionConfiguration -timeout 180m
=== RUN   TestAccImageBuilderDistributionConfigurationDataSource_arn
=== PAUSE TestAccImageBuilderDistributionConfigurationDataSource_arn
=== RUN   TestAccImageBuilderDistributionConfiguration_basic
=== PAUSE TestAccImageBuilderDistributionConfiguration_basic
=== RUN   TestAccImageBuilderDistributionConfiguration_disappears
=== PAUSE TestAccImageBuilderDistributionConfiguration_disappears
=== RUN   TestAccImageBuilderDistributionConfiguration_description
=== PAUSE TestAccImageBuilderDistributionConfiguration_description
=== RUN   TestAccImageBuilderDistributionConfiguration_distribution
=== PAUSE TestAccImageBuilderDistributionConfiguration_distribution
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_amiTags
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_amiTags
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_description
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_description
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_kmsKeyID
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_kmsKeyID
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionAMIDistributionLaunchPermission_userGroups
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionAMIDistributionLaunchPermission_userGroups
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionAMIDistributionLaunchPermission_userIDs
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionAMIDistributionLaunchPermission_userIDs
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionAMIDistributionLaunchPermission_organizationARNs
    acctest.go:697: skipping tests; this AWS account must not be an existing member of an AWS Organization
--- SKIP: TestAccImageBuilderDistributionConfiguration_DistributionAMIDistributionLaunchPermission_organizationARNs (1.17s)
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionAMIDistributionLaunchPermission_ouARNs
    acctest.go:697: skipping tests; this AWS account must not be an existing member of an AWS Organization
--- SKIP: TestAccImageBuilderDistributionConfiguration_DistributionAMIDistributionLaunchPermission_ouARNs (0.06s)
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_name
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_name
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_targetAccountIDs
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_targetAccountIDs
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionContainerDistribution_containerTags
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionContainerDistribution_containerTags
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionContainerDistribution_description
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionContainerDistribution_description
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionContainerDistribution_targetRepository
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionContainerDistribution_targetRepository
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_enabled
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_enabled
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_launchTemplate
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_launchTemplate
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_maxParallelLaunches
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_maxParallelLaunches
=== RUN   TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_snapshotConfiguration
=== PAUSE TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_snapshotConfiguration
=== RUN   TestAccImageBuilderDistributionConfiguration_Distribution_launchTemplateConfiguration
=== PAUSE TestAccImageBuilderDistributionConfiguration_Distribution_launchTemplateConfiguration
=== RUN   TestAccImageBuilderDistributionConfiguration_Distribution_licenseARNs
=== PAUSE TestAccImageBuilderDistributionConfiguration_Distribution_licenseARNs
=== RUN   TestAccImageBuilderDistributionConfiguration_tags
=== PAUSE TestAccImageBuilderDistributionConfiguration_tags
=== RUN   TestAccImageBuilderDistributionConfigurationsDataSource_filter
=== PAUSE TestAccImageBuilderDistributionConfigurationsDataSource_filter
=== CONT  TestAccImageBuilderDistributionConfigurationDataSource_arn
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionContainerDistribution_containerTags
--- PASS: TestAccImageBuilderDistributionConfigurationDataSource_arn (19.34s)
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_description
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionContainerDistribution_containerTags (31.59s)
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_targetAccountIDs
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_description (31.22s)
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_name
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_targetAccountIDs (30.46s)
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionAMIDistributionLaunchPermission_userIDs
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_name (29.85s)
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionAMIDistributionLaunchPermission_userGroups
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionAMIDistributionLaunchPermission_userIDs (31.42s)
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_kmsKeyID
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionAMIDistributionLaunchPermission_userGroups (18.47s)
=== CONT  TestAccImageBuilderDistributionConfiguration_description
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_kmsKeyID (33.65s)
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_amiTags
--- PASS: TestAccImageBuilderDistributionConfiguration_description (32.05s)
=== CONT  TestAccImageBuilderDistributionConfiguration_distribution
--- PASS: TestAccImageBuilderDistributionConfiguration_distribution (22.15s)
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_enabled
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionAMIDistribution_amiTags (32.04s)
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_maxParallelLaunches
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_enabled (31.32s)
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_launchTemplate
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_maxParallelLaunches (31.34s)
=== CONT  TestAccImageBuilderDistributionConfiguration_disappears
--- PASS: TestAccImageBuilderDistributionConfiguration_disappears (14.63s)
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionContainerDistribution_targetRepository
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_launchTemplate (34.80s)
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_snapshotConfiguration
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionContainerDistribution_targetRepository (29.49s)
=== CONT  TestAccImageBuilderDistributionConfiguration_tags
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionFastLaunchConfiguration_snapshotConfiguration (31.43s)
=== CONT  TestAccImageBuilderDistributionConfiguration_Distribution_licenseARNs
--- PASS: TestAccImageBuilderDistributionConfiguration_tags (43.77s)
=== CONT  TestAccImageBuilderDistributionConfigurationsDataSource_filter
--- PASS: TestAccImageBuilderDistributionConfiguration_Distribution_licenseARNs (35.38s)
=== CONT  TestAccImageBuilderDistributionConfiguration_Distribution_launchTemplateConfiguration
--- PASS: TestAccImageBuilderDistributionConfigurationsDataSource_filter (16.89s)
=== CONT  TestAccImageBuilderDistributionConfiguration_basic
--- PASS: TestAccImageBuilderDistributionConfiguration_basic (17.20s)
=== CONT  TestAccImageBuilderDistributionConfiguration_DistributionContainerDistribution_description
--- PASS: TestAccImageBuilderDistributionConfiguration_Distribution_launchTemplateConfiguration (47.84s)
--- PASS: TestAccImageBuilderDistributionConfiguration_DistributionContainerDistribution_description (31.46s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/imagebuilder	348.855s
```

**Comment by ewbankkit** [comment]
@kamilturek Thanks for the contribution :tada: :clap:.

**Comment by ewbankkit** [comment]
```console
% make providerlint
==> Checking source code with providerlint...
```

**Pull request merged**